src/paging.rs

```diff
@@ -539,6 +555,35 @@ impl<T: Translation> PageTableWithLevel<T> {
         }
     }
 
+    /// Modifies a range of page table entries by applying a function to each page table entry.
+    fn modify_range(
+        &mut self,
+        translation: &T,
+        range: &MemoryRegion,
+        f: &PteUpdater,
+    ) -> Result<(), MapError> {
+        let level = self.level;
+        for chunk in range.split(level) {
+            let base_va = chunk.0.start.0;
+            let entry = self.get_entry_mut(chunk.0.start);
+            f(
+                &Range {
+                    start: base_va,
+                    end: base_va + granularity_at_level(level),
+                },
+                entry,
+                level,
+            )
+            .map_err(|_| MapError::PteUpdateFault(*entry))?;
+            if level < LEAF_LEVEL && entry.is_table_or_page() {
```

Comment:
The `subtable` method already checks exactly these conditions, so you don't need to check them again here.

Reply:
Done.